vaadin-dashboard-flow-parent/vaadin-dashboard-flow-integration-tests/src/test/java/com/vaadin/flow/component/dashboard/tests/DashboardIT.java

```diff
@@ -127,6 +160,22 @@ public void changeMaximumColumnCountToNull_widgetsShouldBeOnTheSameRow() {
         Assert.assertEquals(yOfWidget1, widgets.get(1).getLocation().getY());
     }
 
+    @Test
+    public void setDashboardEditable_removeButtonIsVisible() {
+        var widget = dashboardElement.getWidgets().get(0);
+        Assert.assertFalse(isRemoveButtonVisible(widget));
+        clickElementWithJs("toggle-editable");
+        Assert.assertTrue(isRemoveButtonVisible(widget));
+    }
+
+    @Test
+    public void setDashboardNotEditable_removeButtonNotVisible() {
+        clickElementWithJs("toggle-editable");
+        clickElementWithJs("toggle-editable");
+        Assert.assertFalse(
+                isRemoveButtonVisible(dashboardElement.getWidgets().get(0)));
+    }
+
```

Comment:
nit: The web component tests are already testing that the remove button is hidden on non-editable mode and vice-versa

Reply:
Removed all tests that test the visibility of edit buttons.